+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+
+                sum_h = 0;
+                for jj=1:opts.N_finite_elements(ii)
+                    if obj.opts.use_fesd
+                        h = obj.w.h(ii,jj);
+                        sum_h = sum_h + h;
+                    elseif opts.time_optimal_problem && ~opts.use_speed_of_time_variables
+                        h = obj.w.T_final()/(opts.N_stages*opts.N_finite_elements(ii));
+                    else
+                        h = h0;
+                    end
+                    switch opts.rk_representation
+                      case RKRepresentation.integral
+                        % In integral representation stage variables are states.
+                        x_ij_end = opts.D_rk(1)*x_prev;
+                        for kk=1:opts.n_s
+                            x_ijk = obj.w.x(ii,jj,kk);
+                            z_ijk = obj.w.z(ii,jj,kk);
+                            lambda_ijk = obj.w.lambda(ii,jj,kk);
+
+                            fj = s_sot*dcs.f_x_fun(x_ijk, z_ijk, lambda_ijk, ui, v_global, p);
+                            qj = s_sot*dcs.f_q_fun(x_ijk, z_ijk, lambda_ijk, ui, v_global, p);
+                            xk = opts.C_rk(1, kk+1) * x_prev;
+                            for rr=1:opts.n_s
+                                x_ijr = obj.w.x(ii,jj,rr);
+                                xk = xk + opts.C_rk(rr+1, kk+1) * x_ijr;
+                            end
+                            obj.g.dynamics(ii,jj,kk) = {h * fj - xk};
+                            obj.g.z(ii,jj,kk) = {dcs.g_z_fun(x_ijk, z_ijk, ui, v_global, p)};
+                            obj.g.c_lb(ii,jj,kk) = {dcs.c_fun(x_ijk, z_ijk, v_global, p), 0, inf};
+
+                            x_ij_end = x_ij_end + opts.D_rk(kk+1)*x_ijk;
+                            
+                            if opts.g_path_at_stg
+                                obj.g.path(ii,jj,kk) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                            end
+                            if size(model.G_path, 1) > 0
+                                G_path = dcs.G_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                H_path = dcs.H_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                obj.G.path(ii,jj,kk) = {G_path};
+                                obj.H.path(ii,jj,kk) = {H_path};
+                            end
+                            if opts.cost_integration
+                                % also integrate the objective
+                                obj.f = obj.f + opts.B_rk(kk+1)*h*qj;
+                            end
+                        end
+                        if ~opts.right_boundary_point_explicit
+                            x_ijk = obj.w.x(ii,jj,opts.n_s+1);
+                            z_ijk = obj.w.z(ii,jj,opts.n_s+1);
+                            lambda_ijk = obj.w.lambda(ii,jj,opts.n_s+1);
+                            mu_not_c_ij = obj.w.mu_not_c(ii,jj);
```

Comment:
i am confused by the names _not_ 
please add some comments and elaborate

Reply:
Removed, per rbp discussion.